WalletWasabi/Tor/TorProcessManager.cs

```diff
@@ -116,6 +122,8 @@ public async Task<bool> StartAsync(CancellationToken token = default)
 				}
 
 				Logger.LogInfo("Tor is running.");
+				TorControlClient = await InitTorControlOrThrowAsync(token).ConfigureAwait(false);
```

Comment:
Adding `if (_disposed)` to the above while cycle?

Reply:
I'm thinking about this along these lines:

* `TorProcessManager` is supposed to behave as a service that runs over the entire WW application lifetime. (Initialized in Global and disposed in Global).
* `StopAsync` should be renamed to `DisposeAsync` and `TorProcessManager` should implement `IAsyncDisposable` interface. This is because then the intention is clearer that you just start `TorProcessManager` once and dispose it also once.

So this brings us to those `ThrowIfDisposed` calls. We can add such call to other places but I'm not sure if it helps anything because the result is typically the same - an exception. However, the correct solution here is IMO that you `StopAsync` actually waits until `StartAsync` finishes (e.g. async lock).

X) I've just noticed that `StartAsync` does not clean up when cancelation is requested. The better behavior here would be: either Tor is fully initialized, or we make sure to dispose created resources when the operation cancellation is requested. This is to avoid some partial state when "Tor is sort of still running but it shouldn't at all.